addons/html_editor/static/tests/embedded_components.test.js

```diff
@@ -838,6 +839,27 @@ describe("Mount processing", () => {
         expect.verifySteps(["unknown handled"]);
         expect(getContent(el)).toBe(`<div data-embedded="unknown"><p>UNKNOWN</p></div>`);
     });
+    test("Mount a component via a plugin that modifies the Component's env", async () => {
+        let setSelection;
+        class EmbeddedCounter extends Counter {
+            static template = xml`
+                <span class="counter" t-on-click="increment">
+                    <t t-esc="state.value"/>
+                </span>
+            `;
+            setup() {
+                super.setup();
+                setSelection = this.env.setSelection;
+            }
+        }
+        const { plugins } = await setupEditor(`<div>[]a</div>`, {
```

Comment:
you can call `getConfig` before the `setupEditor`, and add the `SimplePlugin` to `config.Plugins` right after